### DISCUSSION TITLE
Convert vaex integer column to timedelta

### DISCUSSION THREAD

**Discussion by DanielvdC** [vaexio/vaex#1740] [Q&A]
Hi,

Is it possible to convert a Vaex dataframe column from string to a timedelta?

I know  `df.date.astype('datetime64')`  exists, but that doesn't work for time strings like `15:38:13`. Is there maybe something like `df.date.astype('timedelta[64ns]')` ?

Any suggestions are greatly appreciated.

**Comment by JovanVeljanoski** [comment]
Hi, 

Ah that seems to be missing, but I think it should very simple to add.. 
let me try and do it!

**Reply by JovanVeljanoski** [reply]
You can track the progress here: https://github.com/vaexio/vaex/pull/1741

Thanks!

**Reply by DanielvdC** [reply]
Awesome, appreciate the fast work! Only got acquainted with Vaex a little while ago, but it really works like a charm. Keep up the good work 😃

**Comment by DanielvdC** [comment]
I am checking if I can use the new function to create a column with datetime, like this:

`df['date'] = df.date_old.astype('datetime64') + df.time.astype('timedelta64[s]')`

where df.date_old is of dtype datetime64[ns] and df.time is of dtype integer.

But this generates the following error:

ValueError: No type alias for timedelta64[s]

Any thoughts?

**Reply by JovanVeljanoski** [reply]
Sure, but if `date_old` is already of type datetime, you don't need to cast it again. Look at this example:
![image](https://user-images.githubusercontent.com/18574951/144572117-9fbb3ca5-59ee-449b-b502-6e24d4b1bf0d.png)

I hope this helps.
